src/masternodes/masternodes.cpp

```diff
@@ -576,10 +595,34 @@ Res CRewardsHistoryStorage::AddBalance(CScript const & owner, DCT_ID poolID, uin
 bool CRewardsHistoryStorage::Flush()
 {
     if (acindex) {
+        auto wallets = GetWallets();
         for (const auto& diff : diffs) {
+            bool isMine = false;
             const auto& pair = diff.first;
-            SetRewardHistory({pair.first, height, pair.second}, diff.second);
+            for (auto wallet : wallets) {
```

Comment:
Can use `for (const auto& wallet : wallets)`

Reply:
Why that's shared_ptr?